src/dbus_api/api.rs

```diff
@@ -252,26 +216,25 @@ fn create_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
     let return_sig = "(os)";
     let default_return = MessageItem::Array(vec![], return_sig.into());
 
-    let pool_name = dbus_try!(
-        object_path_to_pool_name(dbus_context, object_path);
+    let pool_uuid = dbus_try!(
+        object_path_to_pool_uuid(dbus_context, object_path);
         default_return; return_message);
 
     let mut b_engine = dbus_context.engine.borrow_mut();
-    let ref mut pool = get_pool!(b_engine; pool_name; default_return; return_message);
+    let ref mut pool = get_pool!(b_engine; pool_uuid; default_return; return_message);
 
     let specs = filesystems.map(|x| (x.0, x.1, tuple_to_option(x.2).map(|x| Bytes(x))))
         .collect::<Vec<(&str, &str, Option<Bytes>)>>();
     let result = pool.create_filesystems(&specs);
 
     let msg = match result {
```

Comment:
style: so many `let`s... maybe unavoidable but I think you can at least do without `result` since it's only used in the `match`.